src/providers/image_provider.py

```diff
@@ -1,13 +1,13 @@
+import cv2
+
 from dataclasses import dataclass
 from nptyping import NDArray, Shape, UInt8
-from typing import Callable
 
 
 @dataclass(frozen=True)
 class ImageProvider:
     path_to_image: str
-    getter: Callable[[str], NDArray[Shape["*, *, 3"], UInt8]]
 
     @property
     def image(self) -> NDArray[Shape["*, *, 3"], UInt8]:
-        return self.getter(self.path_to_image)
+        return cv2.imread(self.path_to_image, cv2.IMREAD_COLOR)
```

Comment:
Maybe then better call it color_image_provider? Not to embarrass someone, because there could be color images